packages/components/src/core/Tag/index.testUtils.tsx

```diff
@@ -0,0 +1,285 @@
+import { CheckCircleOutline, WbSunny } from "@mui/icons-material";
+import { Args } from "@storybook/react";
+import React from "react";
+import Icon from "../Icon";
+import RawTag from "./index";
+import { ExtraTagProps } from "./style";
+
+const SDS_STYLES: ExtraTagProps["sdsStyle"][] = ["rounded", "square"];
+const SDS_TYPES: ExtraTagProps["sdsType"][] = ["primary", "secondary"];
+
+const DISPLAY_CONTENTS: React.CSSProperties = {
+  display: "contents",
+};
+
+const FONT_WEIGHT_NORMAL: React.CSSProperties = {
+  fontWeight: "normal",
+};
+
+const MID_LABEL: React.CSSProperties = {
+  ...FONT_WEIGHT_NORMAL,
+  borderStyle: "solid none none none",
+  gridColumn: "1 / 6",
+  justifySelf: "stretch",
+  paddingTop: 10,
+};
+
+export const AVAILABLE_ICON_OPTIONS = [
+  undefined,
+  <Icon sdsSize="l" sdsIcon="checkCircle" sdsType="button" />,
+  <Icon sdsSize="l" sdsIcon="loading" sdsType="button" />,
+  <WbSunny />,
+  <CheckCircleOutline />,
+];
+
+const iconOptions = AVAILABLE_ICON_OPTIONS.slice(0, 2);
+
+export const HOVER_OPTIONS = [true, false];
+
+export function CommonScreenshotTestDemo({
+  props,
+  colors,
```

Comment:
Well one other difference is that Gray is the only one whose `sdsType` is split between the main `ScreenshotTest` (gets gray x secondary) and its own `GrayPrimaryScreenshotTest` (gets gray x primary). I realized that some of the secondary Tags (with state of hover) for Success and Warning were also failing a11y test, so I put both primary and secondary of Success and Warning into what had been called `SuccessWarningPrimaryScreenshotTest` and renamed it to `SuccessWarningScreenshotTest` (without `Primary` in the name now).

Reply:
Ah I see! So my refactored `GrayPrimaryScreenshotTestDemo` is wrong, because it iterates with gray + primary and gray + secondary, and we only want gray + primary?

If so, it sounds like we need to add `styles` prop to this component too?